MobileNav/SDL/SDL UI/Map/MapManager.swift

```diff
@@ -0,0 +1,130 @@
+    private var mapView: MGLMapView! {
+        didSet{
+            mapViewCenterPoint = mapView.center
+        }
+    }
+
+    private var mapViewCenterPoint: CGPoint! = .zero
+    var newMapCenterPoint: CGPoint = .zero
+    var mapZoomLevel: Double = 0.0
+
+    func setupMapView(with mapView: MGLMapView, location:CLLocation) {
+        self.mapView = mapView
+
+        mapView.scaleBar.isHidden = true
+        mapView.compassView.isHidden = false
+
+        mapView.isZoomEnabled = true
+        mapView.isScrollEnabled = true
+        mapView.isRotateEnabled = false
+        mapView.isPitchEnabled = false
+
+        let coordinate = CLLocationCoordinate2DMake(location.coordinate.latitude, location.coordinate.longitude)
+        mapView.setCenter(coordinate, zoomLevel: 12, animated: false)
+
+        newMapCenterPoint = mapView.center
+        mapZoomLevel = mapView.zoomLevel
+
+        if NotificationQueue.shared.lastNotification != nil {
+            NotificationCenter.default.post(name: .centerMapOnPlace, object: nil)
+        }
+    }
+
+    func updateScreen() {
+        let mapCenterPoint = self.mapViewCenterPoint
+        let newMapCenterPoint = self.newMapCenterPoint
+        guard mapCenterPoint != newMapCenterPoint else { return }
+        
+        
+        // Set new center
+        let newMapCenterCoordinate = self.mapView.convert(newMapCenterPoint, toCoordinateFrom: self.mapView)
+        self.mapView.setCenter(newMapCenterCoordinate, animated: false)
+        self.newMapCenterPoint = self.mapView.center
+        self.mapView.zoomLevel = self.mapZoomLevel
+    }
+
+    func zoomIn() {
+        mapView.setZoomLevel(mapView.zoomLevel + 1, animated: true)
+    }
+
+    func zoomOut() {
+        mapView.setZoomLevel(mapView.zoomLevel - 1, animated: true)
+    }
+
+    func centerLocation(lat:CLLocationDegrees, long:CLLocationDegrees) {
+        self.mapView.setCenter(CLLocationCoordinate2DMake(lat, long), animated: false)
+    }
+
+}
+
+extension MapManager {
+    var mapManagerTouchHandler: TouchHandler? {
+        return { [unowned self] (touchPoint: CGPoint, scale: CGFloat?, touchType: TouchType) in
+            switch touchType {
+            case .singleTap:
+                self.zoom(to: touchPoint)
+            case .doubleTap:
+                self.zoom(out: touchPoint)
+            case .panMoved:
+                self.panMoved(displacement: touchPoint)
+            case .pinchMoved:
+                self.pinchMoved(centerPoint: touchPoint, scale: scale ?? 1.0)
+            case .panStarted, .pinchStarted: break
+            case .panEnded, .pinchEnded: break
+            }
+            self.updateScreen()
+        }
+    }
+
+    /// Zooms-in and centers around the tap gesture point.
+    ///
+    /// - Parameters:
+    ///   - touchPoint: The location of the tap on the screen
+    private func zoom(to touchPoint: CGPoint) {
+        newMapCenterPoint = touchPoint
+        let newZoomLevel = mapView.zoomLevel + 1
```

Comment:
I believe this happens when the app fails to find the user location. Since this is a test application to use as reference, I decided to set a fake user location should the app fail to grab the real one.